test/integration/good/code-gen/cpp.expected

```diff
@@ -20110,6 +20110,573 @@ stan::model::model_base& new_model(
 
 
 
+  $ ../../../../../install/default/bin/stanc --print-cpp param-constraint.stan
+
+// Code generated by %%NAME%% %%VERSION%%
+#include <stan/model/model_header.hpp>
+namespace param_constraint_model_namespace {
+
+
+inline void validate_positive_index(const char* var_name, const char* expr,
+                                    int val) {
+  if (val < 1) {
+    std::stringstream msg;
+    msg << "Found dimension size less than one in simplex declaration"
+        << "; variable=" << var_name << "; dimension size expression=" << expr
+        << "; expression value=" << val;
+    std::string msg_str(msg.str());
+    throw std::invalid_argument(msg_str.c_str());
+  }
+}
```

Comment:
Is this generated by the compiler? One little optimization trick we use for error checking in the math lib is to tell the compiler that these errors are on the cold path so the compiler/cpu knows there is a very low chance it ever has to care about this code

```suggestion
inline void validate_positive_index(const char* var_name, const char* expr,
                                    int val) {
  if (val < 1) {
      [&]() STAN_COLD_PATH {
        std::stringstream msg;
        msg << "Found dimension size less than one in simplex declaration"
          << "; variable=" << var_name << "; dimension size expression=" << expr
          << "; expression value=" << val;
        std::string msg_str(msg.str());
        throw std::invalid_argument(msg_str.c_str());
    }();
  }
}
```

Reply:
Yes this is generated by the compiler and should be moved to Stan Math (mentioned in https://github.com/stan-dev/stanc3/issues/605). These are variants of validate_non_negative_index as Niko notes in the issue.

I think we should just move this to Math and optimize it there.

Reply:
Wait can't this just use `stan::math::check_positive()`? Like 
```cpp
stan::math::check_positive("", "X", "X", X);
```

Reply:
Could be. I honestly havent looked into that in detail. It should be moved out of generated C++ in any case. Just not sure if we have time to do that before the release.